console/console-init/ui/src/modules/device/dialogs/CreateDevice/MetaDataRow.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useState } from "react";
+import {
+  Grid,
+  GridItem,
+  InputGroup,
+  Button,
+  DropdownPosition,
+  TextInput,
+  DropdownItem,
+  SelectOptionObject
+} from "@patternfly/react-core";
+import { TypeAheadSelect, DropdownWithToggle } from "components";
+import { PlusIcon, MinusCircleIcon } from "@patternfly/react-icons";
+
+export interface IMetaDataRow {
+  onPropertySelect: (e: any, selection: SelectOptionObject) => void;
+  onChangePropertyInput?: (value: string) => Promise<any>;
+  onPropertyClear: () => void;
+  propertySelected?: string;
+  propertyInput?: string;
+  setPropertyInput?: (value: string) => void;
+  formData: any;
+  setFormData: (formData: any) => void;
+}
+
+export const MetaDataRow: React.FC<IMetaDataRow> = ({
+  onPropertySelect,
+  onChangePropertyInput,
+  onPropertyClear,
+  propertySelected,
+  propertyInput,
+  setPropertyInput,
+  formData,
+  setFormData
+}) => {
+  const [currentValue, setCurrentValue] = useState("");
+  const [currentKey, setCurrentKey] = useState("");
+
+  const handleAddChildClick = (event: any, parentKey: string) => {
```

Comment:
i would like to suggest function name as handleAddChildRow or handleAddChildItem etc.

Reply:
Done